pre-docs/writing_yaml_plans.md

```diff
@@ -0,0 +1,294 @@
+# Writing plans in YAML
+
+Plans written in YAML consist of a list of steps executed in order, which allows for complex workflows to be defined. Steps can contain embedded Puppet code expressions to add logic.
+
+## Naming plans
+
+Plan files should be named `planname.yaml`. Note that the file extension must be `.yaml` and not `.yml`.
+
+See [Naming plans](writing_plans.md) for details about where plan files should be located.
+
+## Plan structure
+
+A plan is a YAML document containing a map with several keys:
+
+`steps`: The list of steps to perform
+`parameters`: The parameters accepted by the plan (optional)
+`return`: The value to return from the plan (optional)
+
+## Parameters
+
+A plan can accept a set of parameters, specified in the `parameters` key. The value of `parameters` is a map, where each key is the name of a parameter and the value is a map describing the parameter. A parameter can have a `type`, a `default`, and a `description`.
+
+For example, this plan accepts a `load_balancer` name as a string, and two sets of nodes called `frontends` and `backends`. It then runs a series of tasks to remove nodes from the load balancer, update the application, and then add them back to the load balancer.
+
+```yaml
+parameters:
+  load_balancer:
```

Comment:
Maybe add an example of and empty parameter.

Reply:
what do you mean by empty parameter? one without a type/description?

Reply:
yeah. If the goal is to get started quickly we should show people how to write the simplest possible plan

Reply:
I changed `frontends` to not have a type or description